cms/models.py

```diff
@@ -52,241 +58,576 @@ def can_create_at(cls, parent):
+        return super().route(request, path_components)
+
 
-class CourseIndexPage(CourseObjectIndexPage, Page):
+class CourseIndexPage(CourseObjectIndexPage):
     """
     A placeholder page to group all the courses under it as well
     as consequently add /courses/ to the course page urls
     """
 
     slug = "courses"
 
+    def get_child_by_readable_id(self, readable_id):
+        """Fetch a child page by the related Course's readable_id value"""
+        return self.get_children().get(coursepage__course__readable_id=readable_id)
 
-class ProgramIndexPage(CourseObjectIndexPage, Page):
+
+class ProgramIndexPage(CourseObjectIndexPage):
     """
     A placeholder page to group all the programs under it as well
     as consequently add /programs/ to the program page urls
     """
 
     slug = "programs"
 
-
-class CatalogPage(Page):
-    """
-    A placeholder page object for the catalog page
-    """
-
-    template = "catalog_page.html"
-
-    parent_page_types = ["HomePage"]
-
-    @classmethod
-    def can_create_at(cls, parent):
-        """
-        You can only create one catalog page under the home page.
-        The parent is limited via the `parent_page_type` list.
-        """
-        return (
-            super().can_create_at(parent)
-            and not parent.get_children().type(cls).exists()
-        )
-
-    slug = "catalog"
-
-    def get_context(self, request, *args, **kwargs):
-        """
-        Populate the context with live programs, courses and programs + courses
-        """
-        # Circular import hit when moved to the top of the module
-        from courses.models import CourseRun
-
-        sorted_courserun_qset = CourseRun.objects.order_by("start_date")
-        program_pages = (
-            ProgramPage.objects.live()
-            .order_by("id")
-            .prefetch_related(
-                Prefetch("program__courses__courseruns", queryset=sorted_courserun_qset)
-            )
-        )
-        course_pages = (
-            CoursePage.objects.live()
-            .order_by("id")
-            .prefetch_related(
-                Prefetch("course__courseruns", queryset=sorted_courserun_qset)
-            )
-        )
-        return dict(
-            **super().get_context(request),
-            **get_js_settings_context(request),
-            program_pages=program_pages,
-            course_pages=course_pages,
-            default_image_path=DEFAULT_COURSE_IMG_PATH,
-            hubspot_portal_id=settings.HUBSPOT_CONFIG.get("HUBSPOT_PORTAL_ID"),
-            hubspot_new_courses_form_guid=settings.HUBSPOT_CONFIG.get(
-                "HUBSPOT_NEW_COURSES_FORM_GUID"
-            ),
-        )
+    def get_child_by_readable_id(self, readable_id):
+        """Fetch a child page by the related Program's readable_id value"""
+        return self.get_children().get(programpage__program__readable_id=readable_id)
 
 
-class CourseProgramChildPage(Page):
+class ProductPage(MetadataPageMixin, Page):
```

Comment:
I moved this, `ProgramPage`, and `CoursePage` up in this file because they are very consequential and subject to changes and their previous position was driving me nuts